src/test/ui/issues/issue-31076.rs

```diff
@@ -0,0 +1,15 @@
+#![feature(no_core, lang_items)]
+#![no_core]
+
+#[lang="sized"]
+trait Sized {}
+
+#[lang="add"]
+trait Add<T> {}
+
+impl Add<i32> for i32 {}
+
+fn main() {
+    let x = 5 + 6;
```

Comment:
can you also add a test for `5i32 + 6i32`? Just to make sure that inference doesn't get the better of us.

Reply:
added i32 for the test as well,I assume this is resolved